lualib/lua_util.lua

```diff
@@ -1621,6 +1621,39 @@ local function join_path(...)
 end
 exports.join_path = join_path
 
+---[[[
+--- @function lua_util.url_encode(str)
+-- URL encode string according to RFC 3986
+--
+-- @param str The string to URL encode.
+-- @return The URL encoded string.
+--
+---]]]
+
+local function url_encode (str)
+   str = string.gsub (str, "([^0-9a-zA-Z! '&/:()*._~-])", -- locale independent
+      function (c) return string.format ("%%%02X", string.byte(c)) end)
+   str = string.gsub (str, " ", "+")
+   return str
+end
+exports.url_encode = url_encode
+
+---[[[
+--- @function lua_util.url_decode(str)
+-- URL decode string according to RFC 3986
+--
+-- @param str The string to URL decode.
+-- @return The URL decoded string.
+--
+---]]]
+
+local function url_decode (str)
+   str = string.gsub (str, "+", " ")
+   str = string.gsub (str, "%%(%x%x)", function(h) return string.char(tonumber(h,16)) end)
+   return str
+end
+exports.url_decode = url_decode
```

Comment:
This function exists in `rspamd_util` C library: `rspamd_util.decode_url`.

Reply:
I am not good at C, otherwise I would have proposed the encode function in the same library. I will do some research and upgrade myself, then will propose another commit with amendment to the C library, as well as unit tests for both. 
Thanks, Vsevolod! Always a pleasure to learn from you.